### PR TITLE
Add chef and ohai attributes 2

### LINKED ISSUE #1201: Add chef_version and ohai_version to database tables
This is the task to extract the chef_version and ohai_version out of the uploaded cookbook metadata and to get them into a table.

Displaying the data in the UI comes later (and I actually don't super care about it as a priority)

Getting the chef_version and ohai_version into /universe also comes later (this is what I want to implement on top of this issue).

### PR DESCRIPTION
**Pull request by nellshamrell**
Fixes #1201

### CODE REVIEW

File: app/models/cookbook.rb [hunk 1]
```diff
@@ -258,7 +258,9 @@ def publish_version!(params)
         readme: readme.contents,
         readme_extension: readme.extension,
         changelog: changelog.contents,
-        changelog_extension: changelog.extension
+        changelog_extension: changelog.extension,
+        chef_version: metadata.chef_version,
+        ohai_version: metadata.ohai_version
```
Comment (lamont-granquist) [inline app/models/cookbook.rb]: these should probably be plural `chef_versions` similar to `metadata.dependencies` which is plural even though the DSL method on the metadata.rb is singular "depends"

**Comment by lamont-granquist** [comment]
yeah LGTM, just should be pluralized like the rest of the wire-API is.

**Comment by lamont-granquist** [comment]
I'm not sure where the JSON parsing is handled on the PUT to upload a new cookbook?  There should probably at least be a test around that where you feed it the wire format and watch a correct cookbook version get created in the database...

**Comment by nellshamrell** [comment]
@lamont-granquist - here is where the parsing is handled in the PUT to upload a new cookbook and there is a spec here https://github.com/nellshamrell/supermarket/blob/add-chef-and-ohai-attributes-2/spec/models/cookbook_spec.rb#L503

**Comment by nellshamrell** [comment]
Will get the pluralization in there.

File: spec/models/cookbook_spec.rb [hunk 2]
```diff
@@ -433,7 +433,9 @@ def generate_params(opts = {})
               'yum' => '~> 2.1.3'
             },
             source_url: opts[:source_url],
-            issues_url: opts[:issues_url]
+            issues_url: opts[:issues_url],
+            chef_version: [['12.4.1', '12.4.2'], ['11.2.3', '12.4.3']],
+            ohai_version: [['8.8.1', '8.8.2'], ['8.9.1', '8.9.2']]
```
Comment (lamont-granquist) [inline spec/models/cookbook_spec.rb]: okay, so yeah everything needs to be pluralized, including the wire format here.

Comment (nellshamrell) [inline spec/models/cookbook_spec.rb]: Will get that fixed today.

**Comment by nellshamrell** [comment]
Pluralization is complete.

**Comment by lamont-granquist** [comment]
rubocop angry...

:+1: from me though...

**Comment by lamont-granquist** [comment]
:shipit:

**Comment by robbkidd** [comment]
@nellshamrell Looks great! With the use of JSON columns, we should update [the requirements in the README from PG 9.2+ to 9.3+](https://github.com/chef/supermarket/blob/master/README.md#requirements). (Good news: [PG 9.3 is already the version in the omnibus package](https://github.com/chef/omnibus-supermarket/blob/master/config/projects/supermarket.rb#L31).)

**Comment by nellshamrell** [comment]
Done!

**Comment by robbkidd** [comment]
:+1: :shipit: :rocket: :cake:

**Comment by lamont-granquist** [comment]
coolness

**Pull request merged**